docs/DeveloperGuide.md

```diff
@@ -1,34 +1,384 @@
+
+In addition, the UI contains the different exception and error messages which can be displayed. When a particular exception is thrown (eg. **UNKNOWN_INPUT**), the corresponding function is called in UI to print out the error message (**printUnknownInputMessage()**). 
+
+
+Its main roles are:
+
+* Receiving user commands and returning it to the Main function.
+* Prints visual output in the console for the User
+
+
+### Parser component
+
+
+
+![]()
+*<center/> Figure 1.4 Class diagram of Parser </center> <br/></br>*
+
+
+The Parser component is responsible for decoding the user's input and telling the Main function which command to execute.
+
+It also contains the **programMode** which indicates which mode the program currently is in.
+
+
+It initializes the different commands according to which mode the program currently is in.
+
+It consists of `Parser` and the Command interface classes.
+
+
+Its main roles are:
+
+* Decoding users commands and returning the correct command to the Main function to be executed
+* Catch errors in users commands and return the appropriate exception to the Main function
+* Storing the mode Zoomaster is in (Bookmark/Timetable/Planner)
+
+The interaction of the Parser component with the Command component is covered in greater detail below. 
+
+### Commands component
+Figures 1.3 to 1.6 below show the class-level diagrams for Parser and Command for each different mode. <br/>
+The diagrams are colour coded as such:
+* Orange -> Global
+* Green -> Bookmark Mode
+* Blue -> Timetable Mode
+* Red -> Planner Mode
+
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/mode1.png)
+*<center/> Figure 1.3 Class diagram of Commands valid in all modes (Global) </center> <br/></br>*
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/mode2.png)
+*<center/> Figure 1.4 Class diagram of Commands valid in Bookmark Mode </center> <br/></br>*
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/mode3.png)
+*<center/> Figure 1.5 Class diagram of Commands valid in Timetable Mode</center> <br/></br>*
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/mode4.png)
+*<center/> Figure 1.6 Class diagram of Commands valid in Planner Mode</center> <br/></br>*
+
+The Command component is responsible for carrying out the functions of Zoomaster.
+
+Usually, a successful command will return a message to indicate a successful execution or updates to Zoomaster. Otherwide it will create error messages for the Ui to display to the users.
+
+It consists of `ChangeModeCommand`, `ClearCommand`,  `ExitCommand`,  `HelpCommand`,   `LaunchNowCommand`, `AddBookmarkCommand`,   `DeleteBookmarkCommand`,  `FindBookmarkCommand`,  `LaunchBookmarkCommand`, `ShowBookmarkCommand`, `AddTimetableCommand`, `DeleteTimetableCommand`, `ShowTimetableCommand`, `EditTimetableCommand`, `LaunchTimetableCommand`, `AddMeetingCommand`, `LoadPlannerCommand` and `SavePlannerCommand` classes.
```

Comment:
Do you think this might look better as an unordered list? As it stands, it might seem a bit hard to read, I think a bulleted list might be better.